frontend/src/scenes/settings/settingsLogic.ts

```diff
@@ -214,6 +243,97 @@ export const settingsLogic = kea<settingsLogicType>([
                 }
             },
         ],
+
+        getTextFromTitle: [
+            () => [],
+            (): ((title: JSX.Element | string) => string) => {
+                return (title: JSX.Element | string): string => {
+                    if (typeof title === 'string') {
+                        return title
+                    }
+                    // For JSX elements, we'll extract text content or use a fallback
+                    const children = title?.props?.children
+                    if (typeof children === 'string') {
+                        return children
+                    }
+                    if (Array.isArray(children)) {
+                        // Handle cases where children is an array (e.g., multiple text nodes)
+                        return children.map((child) => (typeof child === 'string' ? child : '')).join('')
+                    }
+                    // Fallback for complex JSX or unexpected cases
+                    return ''
+                }
+            },
+        ],
+
+        filteredLevels: [
```

Comment:
We use fusejs everywhere for this kind of search and its way better than a simpler find in thing.

Would generally recommend swapping to that.

Also, not a massive fan of the whole jsx parsing thing - might be cleaner to add a property to the settings themselves as a proper stringValue (could imagine other UX uses for it anyways)